test/iterable_matchers_test.dart

```diff
@@ -143,13 +143,22 @@ void main() {
         unorderedEquals([3, 1]),
         "Expected: equals [3, 1] unordered "
         "Actual: [1, 2] "
-        "Which: has no match for <3> at index 0");
+        "Which: has no match for <3>");
+    shouldFail(
+        d,
+        unorderedEquals([3, 4]),
+        "Expected: equals [3, 4] unordered "
+        "Actual: [1, 2] "
+        "Which: has no match for any of (<3>, <4>)");
   });
 
-  test('unorderedMatchess', () {
+  test('unorderedMatches', () {
     var d = [1, 2];
     shouldPass(d, unorderedMatches([2, 1]));
     shouldPass(d, unorderedMatches([greaterThan(1), greaterThan(0)]));
+    shouldPass(d, unorderedMatches([greaterThan(0), greaterThan(1)]));
+    shouldPass([2, 1], unorderedMatches([greaterThan(1), greaterThan(0)]));
+    shouldPass([2, 1], unorderedMatches([greaterThan(0), greaterThan(1)]));
```

Comment:
Due to the complexity of how this algorithm actually works, it would be good to add a more exhaustive test as well.

Reply:
what did you have in mind? I think n > 2 and checking every ordering of expected and the values should be exhaustive?

Reply:
More items, some of which overlap with other ones, some of which are only one. Maybe something like...
```
shouldPass([1, 2, 3, 4, 5, 6], unorderedMatches([greaterThan(1), equals(2), allOf([lessThan(3), isNot(0)]), equals(0), predicate((v) => v % 2 == 1), equals(5)]));
```

Reply:
bonus points if you go through all possible orderings of those matchers and test each of them haha

Reply:
Done